fuel/datasets/hdf5.py

```diff
@@ -113,25 +115,86 @@ class H5PYDataset(Dataset):
     """
     ref_counts = defaultdict(int)
 
-    def __init__(self, path, which_set=None, subset=None, load_in_memory=False,
+    def __init__(self, path, which_set, subset=None, load_in_memory=False,
                  flatten=None, driver=None, **kwargs):
         self.path = path
+        self.driver = driver
+        if which_set not in self.available_splits:
+            raise ValueError(
+                "'{}' split is not provided by this dataset".format(which_set))
         self.which_set = which_set
-        self.subset = subset if subset else slice(None, None, None)
+        subset = subset if subset else slice(None, None, None)
+        if subset.step not in (1, None):
+            raise ValueError("subset.step must be either 1 or None")
+        self.subsets = [subset for source in self.provides_sources]
         self.load_in_memory = load_in_memory
         self.flatten = [] if flatten is None else flatten
-        self.driver = driver
-
-        super(H5PYDataset, self).__init__(**kwargs)
-
         for source in self.flatten:
             if source not in self.provides_sources:
-                raise ValueError("trying to flatten source " +
-                                 "'{}' which is not part of ".format(source) +
-                                 "the provided sources")
+                raise ValueError(
+                    "trying to flatten source '{}' which is ".format(source) +
+                    "not provided by the '{}' split".format(self.which_set))
+
+        super(H5PYDataset, self).__init__(**kwargs)
 
         self.load()
 
+    @staticmethod
+    def create_split_array(split_dict):
+        # Determine maximum split, source and string lengths
+        splits = tuple(split_dict.keys())
```

Comment:
Seems like you could use `split_dict` everywhere you use this (since iterating on a dict iterates on keys).